src/logger.ts

```diff
@@ -40,19 +40,38 @@ const transports = [
     : [new winston.transports.Console()]),
 ];
 
+/**
+ * Create a winston.Logger intance with environment defaults.
+ * @deprecated use createLogger instead.
+ * @param metadata default meta data to be added to all logging messages.
+ * @returns winston.Logger
+ */
 export function setLogger(metadata: object = {}): Logger {
-  const releaseSha: string = process.env.RELEASE_SHA || null;
-  const allMetadata: object = {
-    ...metadata,
-    releaseSha,
-  };
+  return createLogger({ defaultMeta: metadata });
+}
 
-  const baseLogger = winston.createLogger({
-    level: level(),
+/**
+ * Create a winston.Logger intance with environment defaults.
+ * @param options winston.LoggerOptions
+ * @returns winston.Logger
+ */
+export function createLogger(options: LoggerOptions | undefined = {}): Logger {
```

Comment:
I don't have strong feels here, but curious why you changed the name?

Reply:
Well, I was trying to sneak in a preference. 

Generally, OOP setters dont return values and if they do it returns `this` when following a builder pattern. Where this function really isnt a setter or a getter. This is a factory function, and the name of the function should be something that makes that fact more obvious.

I know there are always exceptions and changing a name also has its own conversation...